drivers/bbram/bbram_xec.c

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright (c) 2021 Microchip Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT microchip_xec_bbram
+
+#include <drivers/bbram.h>
+#include <errno.h>
+#include <soc.h>
+#include <sys/util.h>
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(bbram, CONFIG_BBRAM_LOG_LEVEL);
+
+#define XEC_VBAT_REGS_ADDR \
+	(DT_REG_ADDR_BY_NAME(DT_INST(0, microchip_xec_pcr), vbatr))
+
+/** Device config */
+struct bbram_xec_config {
+	uintptr_t vbr_base;
+	/** BBRAM base address */
+	uintptr_t base;
+	/** BBRAM size (Unit:bytes) */
+	int size;
+};
+
+static int bbram_xec_check_invalid(const struct device *dev)
+{
+	struct bbram_xec_config const *dcfg =
+		(const struct bbram_xec_config *)dev->config;
+	struct vbatr_regs *const regs = (struct vbatr_regs *)dcfg->vbr_base;
+
+	if (regs->PFRS & BIT(MCHP_VBATR_PFRS_VBAT_RST_POS)) {
+		regs->PFRS |= BIT(MCHP_VBATR_PFRS_VBAT_RST_POS);
+		return 1; /* VBAT power rail failure */
```

Comment:
Errors should be negative and use errno.h. Will `-EFAULT` work here?

Reply:
yes, will update to -EFAULT

Reply:
Fixed in latest commit